wen-restart/src/wen_restart.rs

```diff
@@ -281,18 +291,66 @@ pub(crate) fn find_heaviest_fork(
                     );
                 }
                 return Err(WenRestartError::BlockNotLinkedToExpectedParent(
-                    slot,
+                    *slot,
                     block_meta.parent_slot,
                     expected_parent,
                 )
                 .into());
             }
-            expected_parent = slot;
+            expected_parent = *slot;
         } else {
-            return Err(WenRestartError::BlockNotFound(slot).into());
+            return Err(WenRestartError::BlockNotFound(*slot).into());
+        }
+    }
+    // Now find the hash of the heaviest fork, if block hasn't been replayed, replay to get the hash.
+    let heaviest_fork_slot = expected_parent;
```

Comment:
This will always be equal to `slots.last()` after the filter and sort right? Prefer to move this earlier and be more explicit with that

Reply:
Done.